vignettes/workflows.Rmd

```diff
@@ -1,208 +1,67 @@
+### Configure workflows
 
-This [diagram may help show how the pieces fit together](https://raw.githubusercontent.com/vimc/montagu-machine/master/docs/diagrams/OrderlyWeb%20architecture.png).
+Workflows are defined through a yaml file in a `workflows` directory as `<workflow_name>.yml`
 
-If OrderlyWeb has been deployed onto `orderly.example.com`, then we might declare it in `orderly_config.yml`
-
-```yaml
-remote:
-  default:
-    driver: orderlyweb::orderlyweb_remote
-    args:
-      host: orderly.example.com
-      token: $GITHUB_TOKEN
-```
-
-Once this is done, both Alice and Bob can run reports *on* that server, for example
-
-```r
-orderly::orderly_run_remote("internal-2017-population-TUV-MHL", remote = "science")
+``` {r comment = NA, echo = FALSE}
+dir_tree(path)
 ```
 
-![](workflows_run_remote.gif){width=700px}
-
-this will print to screen the log, just as if it were running locally, though it is in fact running on a fresh R session remotely.  After being run, this report is available to everyone and can be pulled as above
-
-```r
-orderly::orderly_pull_archive("report-a", remote = "default")
+The workflow yaml contains a list of steps of reports to be run in order.
+``` {r results = "asis", echo = FALSE}
+yaml_output(readLines(file.path(path, "workflows", "my_workflow.yml")))
 ```
 
-## Using staging environments
+At the moment workflows can only be run with default parameters for each report.
 
-This can be extended further with the use of "staging environments".  This extends the above (two) scenario so that we have multiple remote orderly archives.  These might be shared file archives, using `orderly::orderly_remote_path` or server hosted archives using `orderlyweb::orderlyweb_remote` or even a combination.
+### Running a workflow
 
-We designate one of these remotes to be our production environment and the others to be staging.  In our work we have used the name pairs `production/science` and `real/testing`; the names used do not matter.
+To run a workflow use `orderly::orderly_workflow`
 
-The `orderly_config.yml` configuration might then look like:
-
-```yaml
-remote:
-  staging:
-    driver: orderlyweb::orderlyweb_remote
-    args:
-      host: orderly.example.com
-      port: 11443
-      token: $GITHUB_TOKEN
-  production:
-    driver: orderlyweb::orderlyweb_remote
-    args:
-      host: orderly.example.com
-      token: $GITHUB_TOKEN
+``` {r}
+out <- orderly::orderly_workflow("my_workflow", root = path)
 ```
 
-Now, as part of the pull request review procedure, the reviewer will run the report on the staging environment, which means that it should work without error on the production environment if they have been set up the same way.
-
-```r
-orderly::orderly_run_remote("report-a", remote = "staging", ref = "origin/myfeature")
+This returns a list containing the name and ID of each of the reports run as part of the workflow
```

Comment:
This isn't true at the moment but will be if  #237 is merged in without any changes